SteamWorkshopService/SteamWorkshopService.cs

```diff
@@ -0,0 +1,283 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using System.Xml.Serialization;
+using System.IO;
+using System.Net;
+using NLog;
+using SteamKit2;
+using System.Net.Http;
+using SteamWorkshopService.Types;
+
+namespace SteamWorkshopService
+{
+    public class WebAPI
+    {
+        private static Logger Log = LogManager.GetLogger("SteamWorkshopService");
+        public const uint AppID = 244850U;
+        public string Username { get; private set; }
+        private string password;
+        public bool IsReady { get; private set; }
+        public bool IsRunning { get; private set; }
+        private TaskCompletionSource<bool> logonTaskCompletionSource;
+
+        private SteamClient steamClient;
+        private CallbackManager cbManager;
+        private SteamUser steamUser;
+
+        private static WebAPI _instance;
+        public static WebAPI Instance
+        {
+            get
+            {
+                return _instance ?? (_instance = new WebAPI());
+            }
+        }
+
+        private WebAPI()
+        {
+            steamClient = new SteamClient();
+            cbManager = new CallbackManager(steamClient);
+
+            IsRunning = true;
+        }
+
+        public async Task<bool> Logon(string user = "anonymous", string pw = "")
+        {
+            if (user == null)
+                throw new ArgumentNullException("User can't be null!");
+            if (user != "anonymous" && pw != "")
+                throw new ArgumentNullException("Password can't be null if user is not anonymous!");
+
+            Username = user;
+            password = pw;
+
+            logonTaskCompletionSource = new TaskCompletionSource<bool>();
+
+            steamUser = steamClient.GetHandler<SteamUser>();
+            cbManager.Subscribe<SteamClient.ConnectedCallback>(OnConnected);
+            cbManager.Subscribe<SteamClient.DisconnectedCallback>(OnDisconnected);
+            cbManager.Subscribe<SteamUser.LoggedOnCallback>(OnLoggedOn);
+            cbManager.Subscribe<SteamUser.LoggedOffCallback>(OnLoggedOff);
+
+            Log.Info("Connecting to Steam...");
+
+            steamClient.Connect();
+
+            await logonTaskCompletionSource.Task;
+            return logonTaskCompletionSource.Task.Result;
+        }
+
+        public void CancelLogon()
+        {
+            logonTaskCompletionSource?.SetCanceled();
+        }
+
+        public async Task<Dictionary<ulong, PublishedItemDetails>> GetPublishedFileDetails(IEnumerable<ulong> workshopIds)
+        {
+            //if (!IsReady)
+            //    throw new Exception("SteamWorkshopService not initialized!");
+
+            using (dynamic remoteStorage = SteamKit2.WebAPI.GetInterface("ISteamRemoteStorage"))
```

Comment:
Space Engineers no longer publishes workshop items using Steam Remote storage.  It's now using UGC, so you need to be using the UGC methods for determining published file details.  One mod that is _definitely_ using UGC is [Sektan's Moon Project](https://steamcommunity.com/sharedfiles/filedetails/?id=1546369205), so you can use that for testing.  Alternatively use any more recent ME mod.

Reply:
Oh boi... Alright, will look into this. However, it could take a while - the Steamkit2 documentation is not very good.

Reply:
So, after testing this, I can confirm that the RemoteStorage interface is still working fine, even with the moon project (at least for the retrieval of meta informationen, which is used in this PR).

I will still add methods for the UGC interface, but it could take more than a week due to the lacking documentation of Steamkit2 and my limited time in the next days.

Reply:
Ok, on the Steamkit2 IRC I have been told, that the WebAPI can still be used to retrieve information  about UGC items (but not to download them). Since I only need to retrieve information in my PR, I propose the following:

1) I'll rename all functions which rely on the WebAPI to reflect that (wherever that's not already the case)
2) If you prefer I can mark the functions which _download_ content as deprectated or remove them altogether
3) I'll create a new PR as soon as I have implemented the UGC methods for downloading content (so they can be used in plugins and the Torch server)

Because I'm starting a new job on Thursday I don't know how much time I will have to work on 3) in the next time (but I'll not abandon it, especially since it's relevant to my own plugin) and I'd rather complete the pending PR before then, if possible (before the code base becomes too outdated, resulting in more an annoying rebase + merge process).

Would you be fine with that solution?
This is especially directed at @Equinox- since you requested the migration to UGC in the first place.

Of course, I'd rebase the PR on latest master before merging :smiley:

Reply:
Done.